### PR TITLE
[stable/mongodb-replicaset]: Updated environment variables in on-star…

### PR DESCRIPTION
**Pull request by scottcrespo**
…t script to access the correct values. Updated README with instructions on how to properly enable authentication.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by scottcrespo** [comment]
#2976 

Updated documentation and fixed environment variables in startup script

**Comment by scottcrespo** [comment]
@unguiculus 

Can you please review?

File: stable/mongodb-replicaset/init/files/on-start.sh [hunk 1]
```diff
@@ -18,8 +18,8 @@ replica_set=$REPLICA_SET
 script_name=${0##*/}
 
 if [[ "$AUTH" == "true" ]]; then
-    admin_user="$ADMIN_USER"
-    admin_password="$ADMIN_PASSWORD"
+    admin_user="$MONGO_INITDB_ROOT_USERNAME"
+    admin_password="$MONGO_INITDB_ROOT_PASSWORD"
```
Comment (unguiculus) [inline stable/mongodb-replicaset/init/files/on-start.sh:22]: Why did you change these? This doesn't make sense.

Comment (scottcrespo) [reply stable/mongodb-replicaset/init/files/on-start.sh:22]: You're right. I dug into things a bit more and it really doesn't make sense.

I grabbed this solution from the thread in #2976 and it "just works." But, it shouldn't be accepted on that basis.

I think the rationale comes from the fact the underlying `mongo` docker image looks for `MONGO_INITDB_ROOT_UESRNAME` and `MONGO_INITDB_ROOT_PASSWORD` to create an initial admin user, and I think [this issue](https://github.com/docker-library/mongo/issues/207) might be pertinent to the problem reported in #2976 

Perhaps `mongodb-replicaset` [on-start.sh](https://github.com/kubernetes/charts/blob/master/stable/mongodb-replicaset/init/files/on-start.sh) is clashing with underlying docker container's [entrypoint script](https://github.com/docker-library/mongo/blob/master/3.4/docker-entrypoint.sh)

That's the best I have to offer so far. Do you have an ideas of the root cause of the problem? Specifically, I'm trying to resolve the issue of the user not being properly created (or I can't log in with the user) when supplying a pre-existing secret to `values.yaml`. 

It's bee some time since I first encountered the problem, so I need to go back and verify I can reproduce the issue. Please let me know your thoughts!

Comment (unguiculus) [reply stable/mongodb-replicaset/init/files/on-start.sh:22]: I'm currently also looking into this. I'm working on a fix that moves all chart-related options out of the config file and passes them to `mongod`. This will make configuration much easier and the user won't have to make changes in multiple places.

Comment (scottcrespo) [reply stable/mongodb-replicaset/init/files/on-start.sh:22]: Awesome! Then I think I should close this PR. Sound good to you? 

I think we both agree this is not the right solution, and it sounds like you're coming up with something much more comprehensive.

**Comment by unguiculus** [comment]
Thanks @scottcrespo
